fhir-validation/src/main/java/com/ibm/fhir/validation/FHIRValidator.java

```diff
@@ -180,8 +201,10 @@ private FHIRValidator() { }
         try {
             evaluationContext.setResolveRelativeReferences(true);
             List<Issue> issues = new ArrayList<>();
-            validateProfileReferences(evaluationContext.getTree().getRoot().asResourceNode(), Arrays.asList(profiles), false, issues);
-            issues.addAll(visitor.validate(evaluationContext, includeResourceAssertedProfiles, profiles));
+            validateProfileReferences(evaluationContext.getTree().getRoot().asResourceNode(), Arrays.asList(profiles), false, issues, failFast);
+            if (!failFast || !hasErrors(issues)) {
```

Comment:
where is the issue added when its failFast?

Reply:
ok, now i get it.  its skipping the visitor.validate if we are in failFast mode and we already have an error.
i think it would be slighlty clearer as `!(failFast && hasErrors(issues))`

Reply:
its now `if (!(failFast && hasErrors(issues)))`